Source/ARTConnection.m

```diff
@@ -284,6 +249,29 @@ - (void)off:(ARTEventListener *)listener {
     [_eventEmitter off:listener];
 }
 
+- (NSString *)createRecoveryKey {
+    // RTN16h
+    if (_state == ARTRealtimeClosing
+        || _state == ARTRealtimeClosed
+        || _state == ARTRealtimeFailed
+        || _state == ARTRealtimeSuspended
+        || (!_key && !_id)) {
+        return nil;
+    }
+    ARTConnectionRecoveryKey *recoveryKey = [[ARTConnectionRecoveryKey alloc] init];
+    recoveryKey.connectionKey = _key;
+    recoveryKey.msgSerial = _latestMessageSerial;
+    
+    NSMutableDictionary *serials = @{}.mutableCopy;
```

Comment:
Odd initialisation, I would keep it simple (literals aren't always the best choice 🤷) ...

```suggestion
    NSMutableDictionary *const serials = [NSMutableDictionary new];
```

Reply:
I would be interested to hear why you think that's not a good idea to create an empty mutable dictionary from an immutable one, which I thought would be more expressive for this case. Happy to apply your suggestion later.